Sources/_StringProcessing/Algorithms/Algorithms/Split.swift

```diff
@@ -309,11 +309,11 @@ extension Collection where Element: Equatable {
   ///   elements.
   @available(SwiftStdlib 5.7, *)
   public func split<C: Collection>(
-    separator: C,
+    pattern: C,
```

Comment:
For `split`, I think calling this parameter `pattern` would make its role unclear. Have you considered `separatorPattern`?